### ISSUE TITLE
no-unused-vars after-used erroring for destructured parameters

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#4620]
Given I am using the `eslint:recommended` preset, which implicitly sets `"no-unused-vars": [2, {"vars": "all", "args": "after-used"}]`:

`([ postSlug, post ]) => post.title` gives me:

```
error  "postSlug" is defined but never used  no-unused-vars
```

I expect this not to error.

`(postSlug, post) => post.title` works as expected (no errors).

ESLint 1.10.3

**Comment by eslintbot** [comment]
@OliverJAsh Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Can you provide more of the code? Is that variable actually used or not?

**Comment by OliverJAsh** [comment]
You can see from the code I posted (a function definition) that the parameter is unused.

``` js
([ postSlug, post ]) => post.title
```

**Comment by nzakas** [comment]
Yes, I was asking if that's the only code or if there's more in the same file.

Also, are you using babel-eslint?

**Comment by nzakas** [comment]
Ooh, I see what's going on. You're using destructuring, which is different than arguments. You could rewrite your code as:

```js
([, post ]) => post.title
```

Unlike arguments, destructuring does not require you to fill in previous values. 

So, I don't believe this is a bug.
